publisher/publisher.go

```diff
@@ -15,15 +17,28 @@ type Publisher struct {
 	specPath    string
 	commandSpec *util.CommandSpec
 	sdAPI       api.API
+	tag         string
+}
+
+func (p *Publisher) tagCommand(specResponse *util.CommandSpec) error {
+	commandFullName := path.Join(specResponse.Namespace, specResponse.Name)
+	promoter, err := promoter.New(p.sdAPI, []string{commandFullName, specResponse.Version, p.tag})
+	if err != nil {
+		return err
+	}
+
+	return promoter.Run()
 }
 
 // Run is a method to publish sdapi and sdstore.
 func (p *Publisher) Run() error {
 	specResponse, err := p.sdAPI.PostCommand(p.commandSpec)
 	if err != nil {
-		return fmt.Errorf("Post failed:%v", err)
+		return fmt.Errorf("Post failed: %v", err)
 	}
 
+	err = p.tagCommand(specResponse)
```

Comment:
Would you check the error here?

Reply:
We talked about tagging failure. When tagging is failed, a command has been uploaded. 
We think that a user may not notice tagging failure if build is passed with tagging failure.  So we choose the implementation which make build fail with tagging failure.